### PR TITLE
fix(presets-fidelity): iso-functional v5 radio-presets conversion (PRESETS-FIDELITY-001)

### PR DESCRIPTION
**Pull request by davidp57**
## Summary

Implements **PRESETS-FIDELITY-001** per [ADR 0003](docs/adr/0003-presets-fidelity.md): make `convert-v5` produce a `presets.yaml` that is iso-functional with the v5 mission's radio presets.

Previously `convert-v5` classified each aircraft only by which preset table its radio `[1]` referenced and assigned a shared preset, **flattening bespoke `["Radio"]` layouts**:
- **Mi-24P** — channel rotation (slot `[1]` = DCS channel 0 holds preset #20, then 1–19) + a second FM radio — both lost.
- **AJS37** — leading dummy (`[1]=0`), 20 UHF + 19 VHF references, 7 hardcoded special frequencies, and a per-channel AM/FM `modulations` table — all lost.

## What changed

- **Converter** (`v5_pipeline_converters.py`): each `radioSettings` entry's radio table is now fully parsed (channels as token refs or hardcoded literals, plus modulations). **Standard** 1:1 layouts keep the lightweight shared assignment. **Bespoke** layouts (rotation, offset, mixing, hardcoded literals, active modulations, or extra radios) get a dedicated `{coalition}_{aircraft}` preset reproducing the exact channel→frequency map (resolving `radioPresets*` tokens, keeping literals) plus modulations.
- **Model** (`presets_manager.py`): `Channel` gains a `mod` field; `RadioDefinition.to_dict()` re-enables the `modulations` table so AM/FM selection round-trips; `from_dict()` reads per-channel `mod`; zero-frequency channels (AJS37 "channel 100") are no longer treated as missing.
- **Regression fixture**: locks the conversion output against the real Tripack `radioSettings.lua`. In-game channel verification is post-merge, fix-forward (per ADR decision).
- **Docs**: `PIPELINE_REFERENCE` (FR/EN) document the per-channel `mod` field and the iso-functional conversion behaviour.

## Quality ratchet

- Dropped `presets_injector.presets_manager` from the mypy `ignore_errors` list and fixed the surfaced type errors.
- Raised `--cov-fail-under` from 64 → 65 (measured ~65.6%).

## Tests

- `poetry run pytest` — 1202 passed, 1 skipped; coverage 65.6%.
- `ruff check` / `ruff format --check` / `mypy src/python/veaf-tools/` — all green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Ensure v5 radio presets conversion remains iso-functional with mission-specific layouts and round-trips per-channel modulation data through the v6 presets model.

New Features:
- Support per-channel modulation metadata on radio channels via an optional mod field that round-trips through presets definitions and images.

Bug Fixes:
- Preserve bespoke v5 radio layouts (rotations, hardcoded channels, extra radios, and per-channel modulations) by generating dedicated per-aircraft presets instead of flattening them to shared presets.

Enhancements:
- Classify v5 radio settings as standard or bespoke and keep standard layouts on shared presets while emitting dedicated presets only for non-standard layouts.
- Improve radio presets conversion by resolving v5 preset table tokens into exact frequencies while retaining hardcoded literals.
- Tighten type hints and error handling in presets management, including more precise return types for serialization methods.

Build:
- Raise the test coverage threshold from 64 to 65 and remove presets_injector.presets_manager from mypy's ignore_errors list.

Documentation:
- Document channel value shapes, including the new per-channel mod field, and describe iso-functional v5 conversion behaviour in the pipeline reference (EN/FR).

Tests:
- Add regression tests against a real Tripack radioSettings.lua fixture to lock Mi-24P and AJS37 bespoke layouts and verify iso-functional conversion.
- Add unit tests covering modulation round-trip on RadioDefinition and PresetDefinition, including zero-frequency channels.

Chores:
- Update backlog and changelog entries to mark PRESETS-FIDELITY-001 as completed and describe the new presets fidelity behaviour.

### CODE REVIEW

File: test/python/mission_builder/test_presets_fidelity.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+class TestTripackPresetsFidelity(unittest.TestCase):
+    """convert_presets must reproduce bespoke v5 radio layouts iso-functionally."""
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls._tmp = tempfile.TemporaryDirectory()
+        out = Path(cls._tmp.name) / "presets.yaml"
+        cls.warnings = convert_presets(_FIXTURE, out)
+        cls.data = yaml.safe_load(out.read_text(encoding="utf-8"))
```
[the remark this answers is not in the document]
Comment (davidp57) [reply test/python/mission_builder/test_presets_fidelity.py:42]: Done in a1c0588: added `test_no_dropped_channel_warnings` asserting the Tripack fixture converts with zero dropped-channel warnings (every preset token resolves, no unsupported expression).

File: src/python/veaf-tools/mission_builder/v5_pipeline_converters.py [hunk 2]
```diff
@@ -121,6 +121,22 @@ def _extract_block_at(text: str, open_brace_pos: int) -> str | None:
...
+#: A parsed channel value: either a resolved reference ``(table_var, token)``
+#: such as ``("radioPresetsBlue", "##RADIO1_20##")`` or a hardcoded frequency.
+_ChannelValue = tuple[str, str] | float
+
+
+@dataclass
+class _RadioSlot:
+    """One DCS radio slot (``["Radio"][N]``) parsed from a ``radioSettings`` entry."""
+
+    block_text: str
```
[the remark this answers is not in the document]
Comment (davidp57) [reply src/python/veaf-tools/mission_builder/v5_pipeline_converters.py]: Done in db80515: `_RadioSlot` now stores the detected `source`, set once in `_parse_radio_settings_entries`; `_emit_dedicated_preset` uses `slot.source` directly — `block_text` and the second detection are gone.
